packages/neo-one-node-blockchain/src/Blockchain.ts

```diff
@@ -643,4 +665,174 @@ export class Blockchain {
       transactions: [],
     });
   }
+
+  private getNep5ChangeSet(applicationsExecuted: readonly ApplicationExecuted[], block: Block): ChangeSet {
+    let transferIndex = 0;
+    const nep5BalancesChanged = new Set<string>();
+    const mutableNep5BalancesChangedKeys: Nep5BalanceKey[] = [];
+    const mutableTransfersSent: Array<{ readonly key: Nep5TransferKey; readonly value: Nep5Transfer }> = [];
+    const mutableTransfersReceived: Array<{ readonly key: Nep5TransferKey; readonly value: Nep5Transfer }> = [];
+
+    applicationsExecuted.forEach((appExecuted) => {
+      if (appExecuted.state === VMState.FAULT) {
+        return;
+      }
+
+      appExecuted.notifications.forEach((notifyEventArgs) => {
+        const { scriptContainer, scriptHash, eventName, state: stateItems } = notifyEventArgs;
+
+        if (stateItems.length === 0) {
+          return;
+        }
+        if (eventName !== 'Transfer') {
+          return;
+        }
+        if (stateItems.length < 3) {
+          return;
+        }
+        if (!stateItems[0].isNull && stateItems[0].type !== StackItemType.ByteString) {
```

Comment:
I think we need to move all of this to another file, very large function and the only blockchain method it needs is `runEngineWrapper`.

Reply:
Also settings, but still easy to do.